model/src/main/java/com/javahelp/model/survey/SurveyResponse.java

```diff
@@ -42,10 +42,7 @@ public Survey getSurvey() {
      * @return {@link SurveyQuestionResponse} for the provided {@link SurveyQuestion}
      */
     public SurveyQuestionResponse getResponse(SurveyQuestion question) {
-        if (!responses.containsKey(question)) {
-            throw new IllegalArgumentException("The SurveyQuestion provided has no response");
-        }
-        return responses.get(question);
+        return responses.getOrDefault(question, new SurveyQuestionResponse(question, -1));
```

Comment:
I like that you use this method in `Map` to simplify the if-else statement.

Reply:
However, I think it's better to specify in the Java doc for this method what -1 means, as I was confused whether -1 means no question found or there is such a question but no response.

Reply:
I think the second `int` parameter is used as a selected index in the whole survey (which is like a collection) to extract that specific response based on the given index. Not completely sure though. It was previously set as `0`, and I think `-1` mean that the index is invalid, as it is out of the scope of the collection `SurveyQuestion`

Reply:
`-1` is just a number I chose to represent situations when this question is not answered by the User. Then, for example, whenever the providerRanker detects `-1` as an answer, it will know not to compare the values. However, I do agree that adding docs is good.